### PR TITLE
Use correct mode for opening device when clearing label

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

**Comment by sonicaj** [comment]
@william-gr should we only do this for boot pool ?

File: libzfs.pyx [hunk 1]
```diff
@@ -2050,6 +2052,8 @@ cdef class ZFSVdev(object):
         if rv != 0:
             raise self.root.get_error()
 
+        clear_label(self.path)
```
Comment (ghost) [inline libzfs.pyx]: If this raises an exception, we will not have written the history entry.

Comment (sonicaj) [reply libzfs.pyx]: Thank you @freqlabs for pointing it out

**Review by ghost — APPROVED** [review]

**Review by william-gr — CHANGES_REQUESTED** [review]
I dont think this should be part of py-libzfs

**Review by william-gr — APPROVED** [review]

**Pull request merged**